### ISSUE TITLE
Datepicker on alert rules screen fails to accept some relative time range configurations

### ISSUE THREAD

**Issue by dpasqualin** [grafana/grafana#36154]
Hi, thank you for great work on grafana 8 :)

I found what seems to be a bug in the time picker on Create Alert Rule page. I want the time range to be "Yesterday", which according to [the docs](https://grafana.com/docs/grafana/latest/dashboards/time-range-controls/
) should be set as `now-1d/d` on both _From_ and _To_ fields. This configuration works in the dashboard time picker control, but not in the Create Alert Rule time picker, which gives me a "Value not in relative time format" error.

This is from the dashboard page, which works.
![valid](https://user-images.githubusercontent.com/68142/123382156-e55d4000-d591-11eb-9401-ea8a46579067.png)

This is from Create Alert Rule page, which doesn't work.
![invalid](https://user-images.githubusercontent.com/68142/123382151-e4c4a980-d591-11eb-8f4c-6f45334e8122.png)

**Environment**:
- Grafana version: 8.0.4
- OS Grafana is installed on: debian stretch

Thanks for your time.

**Comment by Hardosaur** [comment]
Just want to add to this bug: the same error appears when you try to define some future time period, as show on the screenshot attached.
![image](https://user-images.githubusercontent.com/11484509/140276878-238d486e-2beb-4842-9528-d836f8e7a056.png)

Future period is critical to create alerts based on predictions.

**Comment by peterholmberg** [comment]
@Hardosaur alerting does not support predictions right now.

**Comment by peterholmberg** [comment]
After some investigation I have found that we can not support the "Today so far syntax" in alerting. The reason being that we persist timerange in seconds in the backend which would cause a lot of bugs if we added this. We will need to reconsider how we store timeranges before we can support this.

**Comment by gillesdemey** [comment]
We will need to update the backend to support relative time definitions like these, for now we should update the documentation stating that those are not supported in alerting.

**Comment by GrafanaWriter** [comment]
@achatterjee-grafana - FYI

**Comment by visit1985** [comment]
@peterholmberg I don’t see, how adapting the docs solves this issue. In addition there seems to be a missunderstanding in what we want to achieve by using the `now/h` notation. You can read from `now/d` to `now` as „this day so far“. But there are other cases like from `now/d-1d` to `now/d`, which is „the last full day“. Time ranges like this can be important for monitoring SLAs.

**Comment by gillesdemey** [comment]
I've created [a follow-up issue](https://github.com/grafana/grafana/issues/44823) so we don't lose track of the initial request to support `now-n/n` date and time expressions. 

We currently do not have proper support for these expressions in the ruler API and the documentation has been reflected to document that caveat.
